databricks/koalas/indexes.py

```diff
@@ -456,6 +457,47 @@ def to_series(self, name: Union[str, Tuple[str, ...]] = None) -> Series:
                                          column_index_names=None),
                       anchor=kdf)
 
+    def duplicated(self, keep=False):
+        """
+        Indicate duplicate index values.
+        Duplicated values are indicated as ``True`` values in the resulting
+        array. Either all duplicates, all except the first, or all except the
+        last occurrence of duplicates can be indicated.
+
+        Parameters
+        ----------
+        keep : {False}, default False
+            - ``False`` : Mark all duplicates as ``True``.
+
+        Returns
+        -------
+        duplicated : Index
+
+        See Also
+        --------
+        Series.duplicated : Equivalent method on koalas.Series.
+        DataFrame.duplicated : Equivalent method on koalas.DataFrame.
+
+        Examples
+        --------
+        By default, for each set of duplicated values, all duplicates is set to True:
+
+        >>> idx = ks.Index(['lama', 'cow', 'lama'])
+        >>> idx.duplicated()
+        0    True
+        1   False
+        2    True
+        Name: duplicated, dtype: bool
+        """
+        index_scols = self._internal.index_scols
+        sdf = self._internal.sdf
+        window = Window.partitionBy(index_scols).rowsBetween(Window.unboundedPreceding,
+                                                             Window.unboundedFollowing)
+        sdf = sdf.withColumn('duplicated', F.count(index_scols[0]).over(window) > 1)
```

Comment:
after this the order will be shuffled,
so i think how about we make one more extra column to keep natural order like below.
(and use more complex column name for 'duplicated' to avoid duplication with user column)
```python
sdf = sdf.withColumn('__natural_order__', F.monotonically_increasing_id())
sdf = sdf.withColumn('__duplicated__', F.count(index_scols[0]).over(window) > 1)
sdf = sdf.select('__duplicated__').orderBy('__natural_order__')
```

and now we returns `Series`, not `Index`,


(actually i'm not pretty sure is it right way that we returns `Index` rather than `np.ndarray`)

```python
internal = _InternalFrame(sdf=sdf, index_map=[('__duplicated__', None)])
return DataFrame(internal).index
```

Reply:
Yes, there are some functions which we need return result in natural order. But there is no extra column like `__index_level_0__`  in `MultiIndex`.